### PR TITLE
Revamp formatters

### PR DESCRIPTION
**Pull request by jessemyers**
Changes:

 - Clearer/simpler base class implementation
 - Less code in subclasses
 - Tests

### CODE REVIEW

File: microcosm_flask/formatting/base.py [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
-        # Formatting could need the response schema
-        # e.g. to specify column ordering in CSV response
-        self.response_schema = response_schema
 
-    @abstractmethod
-    def make_response(self, response_data, headers):
-        """
-        Build network response object from response data `dict`
-        The user can also pass in additional headers
+    def __call__(self, response_data, headers=None, **kwargs):
```
Comment (jessemyers) [inline microcosm_flask/formatting/base.py]: Refactoring here to delegate to `build_response` and `build_headers`

File: microcosm_flask/tests/formatting/test_json_formatter.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+def test_make_response():
+    graph = create_object_graph(name="example", testing=True)
+    formatter = JSONFormatter()
+
+    with graph.app.test_request_context():
+        response = formatter(dict(foo="bar"))
+
+    assert_that(response.data, is_(equal_to(b'{\n  "foo": "bar"\n}\n')))
+    assert_that(response.content_type, is_(equal_to("application/json")))
```
Comment (jessemyers) [inline microcosm_flask/tests/formatting/test_json_formatter.py]: Note that the existing behavior in `jsonify` does not add an explicit `charset=utf-8` encoding to the content type. This seems weird, but isn't new, so I'm not changing it here.

File: microcosm_flask/formatting/base.py [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+        response = self.build_response(response_data)
+        headers = self.build_headers(headers=headers or {}, **kwargs)
+        response.headers.extend(headers)
```
Comment (netama) [inline microcosm_flask/formatting/base.py]: any point worrying about duplicate headers?

Comment (jessemyers) [reply microcosm_flask/formatting/base.py]: Not really.

**Pull request closed without merging**